test/url.cpp

```diff
@@ -854,6 +854,122 @@ class url_test
             it--;
             BOOST_TEST(it->encoded_key() == "x");
         }
+
+        {
+            url u("http://user:pass@example.com:80/file.txt");
+            auto qp = u.params();
+            BOOST_TEST(qp.empty());
+            qp.insert_encoded(
+                qp.insert_encoded(
+                    qp.insert_encoded(
+                        qp.insert_encoded(
+                            qp.insert_encoded(qp.end(), {"e", "5"} ), {"d", "4"} ), {"c", "3"} ), {"b", "2"} ), {"a", "1"} );
+            BOOST_TEST(!qp.empty());
```

Comment:
The tests should exercise each member function in their order of appearance in the header file. If this order is ever changed, the tests need to be refactored as well. Each test should mention the function in a comment.

See: 
https://github.com/boostorg/json/blob/develop/test/object.cpp#L1050

And then we check the coverage report to make sure that every code path is tested.

I'll note that your test does too much - inserting 5 times should not be needed. You should calculate all of the meaningful permutations of insertions. Example:

Append A to {}
Append B to { A }
Insert A to { B } before B
Insert B to { A, C } before C

and so on. Do each with a non-empty value and an empty value. And encoded versus decoded. So this is a total of 16 tests for `insert`. Each one should be labeled with a comment, like this:

https://github.com/boostorg/json/blob/develop/test/object.cpp#L961

For above, the permutations are "small", "small, existing duplicate", "small, new duplicate" and so forth, for the function with signature `object::insert( initializer_list<...> )` (line 959)

Also note the use of braces to isolate the scope of each test.